### PR TITLE
Parser created for iosxe show snmp user

### PR DESCRIPTION
**Pull request by qduk**

### CODE REVIEW

File: src/genie/libs/parser/iosxe/show_snmp.py [hunk 1]
```diff
@@ -79,3 +79,118 @@ def cli(self, output=None):
 
         return ret_dict
 
+
+
```
Comment (badalsoyantar) [inline src/genie/libs/parser/iosxe/show_snmp.py:83]: Need to remove extra space from top.
Also need to add comment for all new parsers we add

File: src/genie/libs/parser/iosxe/show_snmp.py [hunk 2]
```diff
@@ -79,3 +79,118 @@ def cli(self, output=None):
...
+        # Group-name: group1
+        group_capture = "(?P<group>\S+)"
+        p_group = re.compile("Group-name:\s+{group_capture}".format(group_capture=group_capture))
+
+
+        snmp_user_obj = {}
+        current_loop_user = ""
+
+        for line in output.splitlines():
+            if p_user.match(line):
```
Comment (badalsoyantar) [inline src/genie/libs/parser/iosxe/show_snmp.py:166]: Please add matching line as comment for each regex.

**Comment by badalsoyantar** [comment]
Can you please do 'make json' in genieparser directory. Otherwise when you do device.parse it won't detect the new parser you created.

File: src/genie/libs/parser/iosxe/show_snmp.py [hunk 3]
```diff
@@ -79,3 +79,118 @@ def cli(self, output=None):
...
+
+
+# ===================
+# Schema for:
+#  * 'show snmp user'
+# ===================
+class ShowSnmpUserSchema(MetaParser):
+    """Schema for show snmp user."""
+
+    schema = {
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/show_snmp.py:91]: please add key name for username like:
```
schema = {
    'user_name': {
        Any(): {
            Optional("access_list"): str,

File: src/genie/libs/parser/iosxe/show_snmp.py [hunk 4]
```diff
@@ -79,3 +79,118 @@ def cli(self, output=None):
...
+        Any(): {
+            Optional("access-list"): str,
```
Comment (tahigash) [inline src/genie/libs/parser/iosxe/show_snmp.py:93]: we usually use `_` in key name for iosxe. please change to from `-` to `_`.

**Review by tahigash — CHANGES_REQUESTED** [review]

**Comment by itdependsnetworks** [comment]
@badalsoyantar  make json does not seem to work, and there seems to be some processes that only work locally. 


```
ntcgenie@1caa7cbc41e0:~/genieparser$ make json

--------------------------------------------------------------------
Generating Parser json file

Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "src/genie/json/make_json.py", line 818, in genie.json.make_json.make_genieparser
  File "src/genie/json/make_json.py", line 60, in genie.json.make_json.MakeParsers.__init__
AssertionError
make: *** [Makefile:184: json] Error 1
ntcgenie@1caa7cbc41e0:~/genieparser$ 
```


```
>>> import genie.json.make_json
>>> genie.json.make_json
<module 'genie.json.make_json' from '/usr/local/lib/python3.7/site-packages/genie/json/make_json.cpython-37m-x86_64-linux-gnu.so'>
```

File: src/genie/libs/parser/iosxe/show_snmp.py (hunk 2 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_snmp.py:167]: Matching lines added.

File: src/genie/libs/parser/iosxe/show_snmp.py (hunk 1 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_snmp.py]: Removed extra lines. I also added comments. If I am still missing a comment please let me know and I can update.

File: src/genie/libs/parser/iosxe/show_snmp.py (hunk 3 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_snmp.py]: Schema updated.

File: src/genie/libs/parser/iosxe/show_snmp.py (hunk 4 above)
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_snmp.py]: Replaced hyphens with underscores.

File: src/genie/libs/parser/iosxe/show_snmp.py [hunk 5]
```diff
@@ -79,3 +79,127 @@ def cli(self, output=None):
...
+        # Group-name: group1
+        group_capture = "(?P<group>\S+)"
+        p_group = re.compile("Group-name:\s+{group_capture}".format(group_capture=group_capture))
+
+
+        snmp_user_obj = {}
+        current_loop_user = ""
+
+        for line in output.splitlines():
+            # User name: SNMPv3-alfa
```
[the remark this answers is not in the document]
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_snmp.py:166]: Updated

File: src/genie/libs/parser/iosxe/show_snmp.py [hunk 6]
```diff
@@ -79,3 +79,127 @@ def cli(self, output=None):
...
+        #
+        # User name: nmsops
+        # Engine ID: 00000063000100A20A101B3E
+        # storage-type: nonvolatile	 active	access-list: 69
+        # Authentication Protocol: MD5
+        # Privacy Protocol: DES
+        # Group-name: nmcigroup
+
+        # User name: SNMPv3-alfa
+        user_capture = "(?P<user>\S+)"
```
[the remark this answers is not in the document]
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_snmp.py:134]: I think I saw the separate variable in an older parser that I used as an example. I agree the inline variable is cleaner. Took away separate variable and added inline.

File: src/genie/libs/parser/iosxe/show_snmp.py [hunk 7]
```diff
@@ -79,3 +79,127 @@ def cli(self, output=None):
...
+        p_user = re.compile("User\s+name:\s+{user_capture}".format(user_capture=user_capture))
```
[the remark this answers is not in the document]
Comment (qduk) [reply src/genie/libs/parser/iosxe/show_snmp.py:135]: Added carrot to front and $ to end of all regex

**Review by badalsoyantar — APPROVED** [review]

File: src/genie/libs/parser/iosxe/show_snmp.py (hunk 3 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/show_snmp.py]: confirmed

File: src/genie/libs/parser/iosxe/show_snmp.py (hunk 4 above)
Comment (tahigash) [reply src/genie/libs/parser/iosxe/show_snmp.py]: confirmed

**Review by tahigash — APPROVED** [review]

**Comment by ThomasJRyan** [comment]
Please resolve merge conflicts

**Pull request merged**